frontend_server/server.js

```diff
@@ -130,17 +149,34 @@ app.post('/fetch-ai-insight', async (req, res) => {
       },
     )
 
+    if (aiResponse.status === 429) {
+      RATE_LIMIT_REACHED = true
+      throw new Error('Rate limit reached')
```

Comment:
is this what that other substring check is looking for? maybe we can return the actual error code and check for the specific code ?